harness/determined/common/experimental/determined.py

```diff
@@ -419,39 +419,49 @@ def remove_oauth_client(self, client_id: str) -> None:
     def stream_trials_metrics(
         self, trial_ids: List[int], group: str
     ) -> Iterable[metrics.TrialMetrics]:
-        """
-        Streams metrics for one or more trials sorted by
-        trial_id, trial_run_id and steps_completed.
+        """Get a stream of metrics for this trial.
 
         Arguments:
-            trial_ids: List of trial IDs to get metrics for.
+            trial_ids: The trial IDs to stream metrics for.
+            group: The metric group to stream.  Common values are "validation" and "training", but
+                group can be any value passed to master when reporting metrics during training
+                (usually via a context's `report_metrics`).
+
+        Returns:
+            An iterable of :class:`~determined.experimental.TrialMetrics` objects.
         """
+        if group not in ("training", "validation"):
+            raise ValueError(
+                f"Invalid metric group: {group}. Valid values are 'training' and 'validation'"
+            )
```

Comment:
Docstring is right but we are still hard asserting here

Reply:
Thanks. Took me a minute to realize that this was in both `determined` and `trial`.